libvmi/driver/kvm/include/asm/kvm.h

```diff
@@ -0,0 +1,364 @@
+/* SPDX-License-Identifier: GPL-2.0 WITH Linux-syscall-note */
```

Comment:
Could you explain this comment? Is this header under GPL2? I don't think we can mix GPL and LGPL

Reply:
cc @adlazar, @mdontu

Reply:
In kernel headers, that line specifically means that the header is placed outside of the Linux GPL license requirements, allowing it to be included into non GPL compliant user-space application code.

Anything built with LibVMI would constitute a user-space application falling safely within that exception; HOWEVER, the placement of that header with that marking within LibVMI is inherently problematic. Such headers are meant to be included by reference, not embedded entirely.

Reply:
To elaborate:

If both headers are intended to be included in the installation of a kernel with kvmi, then relicensing is unnecessary as far as I can tell (and embedding of the header within LibVMI should be unnecessary).

The kvmi driver would literally just need to #include the kvmi.h etc headers (which should still retain the "PDX-License-Identifier: GPL-2.0 WITH Linux-syscall-note"), and we're all set AND LibVMI safely remains LGPL.

In addition to the doing the above, if LibVMI-specific material has been added into the headers in question, it should be extracted into a much smaller header, LGPL licensed, and included in the LibVMI kvmi subtree.

Reply:
I agree. I think we can 'unbundle' kvm.h. It's part of the kernel UAPI and hence it should not see changes that break backwards compatibility.

I think we used qemu as example which also bundles them (since 2011). I don't know why the chose to do that.

Reply:
@Wenzel Thank you for addressing